From a9599be2d86dd534f37b46ab38a970ad88619c08 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Fri, 28 Aug 2015 17:25:28 -0700
Subject: [PATCH] OS X: Fix build when system-freetype is detected

We need to pass the correct include paths that configure detected.

Change-Id: I9832b93d3f2a07776f793c92355d764603d48c0d
---
 config.tests/unix/freetype/freetype.pri            | 15 ++++++++++++---
 src/platformsupport/fontdatabases/mac/coretext.pri |  2 +-
 src/plugins/platforms/cocoa/cocoa.pro              |  1 +
 3 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/qtbase/config.tests/unix/freetype/freetype.pri b/qtbase/config.tests/unix/freetype/freetype.pri
index 05299ed..93d09e3 100644
--- a/qtbase/config.tests/unix/freetype/freetype.pri
+++ b/qtbase/config.tests/unix/freetype/freetype.pri
@@ -1,4 +1,4 @@
-!cross_compile {
+!cross_compile:isEmpty(QMAKE_CFLAGS_FONTCONFIG) {
     TRY_INCLUDEPATHS = /include /usr/include $$QMAKE_INCDIR $$QMAKE_INCDIR_X11 $$INCLUDEPATH
     # LSB doesn't allow using headers from /include or /usr/include
     linux-lsb-g++:TRY_INCLUDEPATHS = $$QMAKE_INCDIR $$QMAKE_INCDIR_X11 $$INCLUDEPATH
@@ -10,6 +10,15 @@
 } else {
    # If we are cross-compiling, then there is still a remote possibility that
    # configure detected font-config & freetype,  stored in these variables.
-   QMAKE_CFLAGS += $$QMAKE_CFLAGS_FONTCONFIG
-   QMAKE_CXXFLAGS += $$QMAKE_CFLAGS_FONTCONFIG
+   for(i, QMAKE_CFLAGS_FONTCONFIG) {
+        contains(i, "-I.*") {
+           i ~= s/^-I//
+           INCLUDEPATH += $$i
+        } else {
+            QMAKE_CFLAGS += $$i
+            QMAKE_CXXFLAGS += $$i
+        }
+   }
 }
+
+LIBS += $$QMAKE_LIBS_FONTCONFIG
\ No newline at end of file
diff --git a/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri b/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri
index ebb64d1..3e64f94 100644
--- a/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri
+++ b/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri
@@ -1,7 +1,7 @@
 HEADERS += $$PWD/qcoretextfontdatabase_p.h $$PWD/qfontengine_coretext_p.h
 OBJECTIVE_SOURCES += $$PWD/qfontengine_coretext.mm $$PWD/qcoretextfontdatabase.mm
 
-contains(QT_CONFIG, freetype) {
+contains(QT_CONFIG, freetype)|contains(QT_CONFIG, system-freetype) {
     include($$QT_SOURCE_TREE/src/3rdparty/freetype_dependency.pri)
     HEADERS += $$QT_SOURCE_TREE/src/gui/text/qfontengine_ft_p.h
     SOURCES += $$QT_SOURCE_TREE/src/gui/text/qfontengine_ft.cpp
diff --git a/qtbase/src/plugins/platforms/cocoa/cocoa.pro b/qtbase/src/plugins/platforms/cocoa/cocoa.pro
index 05d0bb4..4298ebb 100644
--- a/qtbase/src/plugins/platforms/cocoa/cocoa.pro
+++ b/qtbase/src/plugins/platforms/cocoa/cocoa.pro
@@ -80,6 +80,7 @@ contains(QT_CONFIG, opengl.*) {
 RESOURCES += qcocoaresources.qrc
 
 LIBS += -framework Cocoa -framework Carbon -framework IOKit -lcups
+CONFIG += qpa/basicunixfontdatabase
 
 QT += core-private gui-private platformsupport-private
 
-- 
2.7.3

